lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -3066,7 +3066,16 @@ void ByteCodeGenerator::EmitOneFunction(ParseNodeFnc *pnodeFnc)
         // to be verified. Ideally if the function has simple parameter list then we can avoid inserting the opcode and the additional call.
         if (pnodeFnc->IsGenerator() && !pnodeFnc->IsModule())
         {
-            EmitDummyYield(this, funcInfo);
+            if (
+                this->GetScriptContext()->GetThreadContext()->IsRuntimeInTTDMode() ||
+                (!CONFIG_ISENABLED(Js::RemoveDummyYieldFlag) || pnodeFnc->HasNonSimpleParameterList()))
+            {
+                EmitDummyYield(this, funcInfo);
+            }
+            else
+            {
+                this->Writer()->Reg1(Js::OpCode::LdUndef, funcInfo->yieldRegister);
```

Comment:
In AsyncGenerators this results in writing undefined to what should be the location of the ResumeYieldData - unfortunately removing it just leaves it as a nullptr which isn't right either - in any event I don't think this instruction is useful.

Within the EmitDummyYield undefined is loaded into the yieldRegister then yielded (as you need to yield something) - whereas the ResumeYield that follows results in ResumeYieldData being set here.